### PR TITLE
InternshipStack service was fixed 🥳

### PR DESCRIPTION
**Pull request by Evgeniy-Sinicin**

### CODE REVIEW

File: InternshipBE/DAL/Repositories/InternshipStackRepository.cs [hunk 1]
```diff
@@ -16,10 +16,10 @@ public InternshipStackRepository(ApplicationDbContext context) : base(context)
...
         public async Task<List<InternshipStack>> GetAllByInternshipIdAsync(int internshipId)
         {
-            return await _context.InternshipStacks.AsNoTracking()
-                                                  .Include(x => x.Internship)
-                                                  .Where(x => x.InternshipId == internshipId)
-                                                  .ToListAsync();
+            return await Task.Run(() => _context.Internships.AsNoTracking()
+                                                            .Include(x => x.InternshipStacks)
+                                                            .FirstOrDefault(x => x.Id == internshipId).InternshipStacks
+                                                            .ToList());
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/InternshipStackRepository.cs]: Why Task.Run? ToListAsync?

File: InternshipBE/BL/Interfaces/IInternshipStackService.cs [hunk 2]
```diff
@@ -6,10 +6,12 @@ namespace BL.Interfaces
 {
     public interface IInternshipStackService
     {
+        Task<List<GetInternshipStackDTO>> GetInternshipStacksAsync();
+
         Task<List<GetInternshipStackDTO>> GetInternshipStacksByInternshipIdAsync(int internshipId);
 
-        Task<FullInternshipStackDTO> CreateInternshipStackAsync(CreateInternshipStackDTO internshipStackDto);
+        Task<GetInternshipStackDTO> CreateInternshipStackAsync(InternshipStackDTO internshipStackDto);
```
Comment (romanstat) [inline InternshipBE/BL/Interfaces/IInternshipStackService.cs]: out GetInternshipStackDTO and in InternshipStackDTO ?

File: InternshipBE/DAL/Repositories/InternshipStackRepository.cs (hunk 1 above)
Comment (Evgeniy-Sinicin) [reply InternshipBE/DAL/Repositories/InternshipStackRepository.cs]: ICollection hasn't ToListAsync() 🙄

File: InternshipBE/BL/Interfaces/IInternshipStackService.cs (hunk 2 above)
Comment (Evgeniy-Sinicin) [reply InternshipBE/BL/Interfaces/IInternshipStackService.cs]: We get dto without id, create entity, and return dto with autogenerated id

**Review by romanstat — APPROVED** [review]

**Pull request merged**